Cover art

micronova-jb opened the pull request:
Updated shairport-sync-metadata-reader for use with AmpliPi. This version releases the picture data in a usable format. Album art from iTunes is now properly sent and translated over AirPlay.

micronova-jb committed:
updated reader to output pictures

-Pictures are saved in the directory you're in when you call the metadata reader. They should save as IMG_A1...A2... and so on

micronova-jb committed:
Print clear image data

-Image name is needed so that other scripts can find where it's saved
-Image length is an easy way to determine if the file is empty (0 bytes) or not

linknum23 commented on shairport-sync-metadata-reader.c:
```diff
@@ -168,15 +169,15 @@ int main(void) {
-                  int inputlength=32768;
+                //if (code!='PICT') {
+                  int inputlength=999999;
                   if (b64size<inputlength)
```
If its too big, it shouldnt even try to decode the image

linknum23 commented on shairport-sync-metadata-reader.c:
```diff
@@ -149,7 +150,7 @@ int main(void) {
         // basically, we need to get hold of the base-64 data, if any
         size_t outputlength=0;
-        char payload[32769];
+        char payload[1000000];
```
Make this a #define MAX_PAYLOAD (1024*1024)
``` char payload[MAX_PAYLOAD+1] ```

linknum23 commented on shairport-sync-metadata-reader.c:
```diff
@@ -252,6 +253,12 @@ int main(void) {
             break;
           case 'PICT':
             printf("Picture received, length %u bytes.\n",length);
```
Add TODO: remove old pictures

micronova-jb replied:
Currently, old pictures are removed whenever you start up a new instance of the same source (i.e. putting Jason's iPhone on channel 3 will remove all prior pictures in channel 3's directory). I will admit that this solution kind of sucks though

micronova-jb replied:
Just thought of another way to handle picture deletion - if iTunes pauses for about 10 seconds, the 'PAUSED' information gets updated in 'currentSong'.
Maybe the files can be deleted at this time? Still isn't perfect, but it would be better than the current implementation.

micronova-jb committed:
payload size cleanup

-Better implementation of picture size in the metadata script
-If pictures are too large, they will be ignored

micronova-jb committed:
Add functionality to delete old cover art

Add a switch/case block to cycle image names, deleting any images associated with the next character in a three character rotation

micronova-jb committed:
Refactor image creation/deletion

Image naming, along with the process of deleting old images, is leaner and more intelligent

micronova-jb commented:
> You may be able to do this with less repetition using %c to print the namechar, and namechar++ to increment to the next character

Does the latest commit look better? It seems to be a lot cleaner now, while making use of %c - it also has been tested/verified on the AmpliPi system

micronova-jb committed:
Update to previous commit

The previous commit didn't have the full set of changes.
